src/main/java/gov/adlnet/xapi/client/StatementClient.java

```diff
@@ -200,9 +200,7 @@ private StatementClient addFilter(String key, String value) {
 				client.filters = new TreeMap<String, String>();
 			}
 			if (this.filters != null) {
-				for (Entry<String, String> filter : filters.entrySet()) {
-					client.filters.put(filter.getKey(), filter.getValue());
-				}
+				client.filters.putAll(this.filters);
```

Comment:
Sorry we're going to keep jxapi to be as spec-compliant as we can for everyone else. Obviously you're free to keep it in your own fork though for use. Thanks for the other tweaks you made.